var/spack/repos/builtin/packages/cgal/package.py

```diff
@@ -59,28 +65,52 @@ class Cgal(CMakePackage):
     depends_on('qt@5:', when='+demos')
 
     # Optional Third Party Libraries
+    depends_on('eigen', when='+eigen')
+
     # depends_on('leda')
     # depends_on('mpfi')
     # depends_on('rs')
     # depends_on('rs3')
     # depends_on('ntl')
-    # depends_on('eigen')
     # depends_on('libqglviewer')
     # depends_on('esbtl')
     # depends_on('intel-tbb')
 
+    # Header only builds became optimal in 4.9, default thereafter
```

Comment:
Did you mean "optional" instead of "optimal"? Btw, you can add a `msg='...'` parameter to the `conflicts()` call.